### ISSUE TITLE
Duplicate pageView on initial page load

### ISSUE THREAD

**Issue by Damil2020** [nuxt-community/gtm-module#36]
Hi folks!
Can anyone help me avoiding a duplicate pageView being generated on the first page load? I have the pageTracking setting enabled, since I need to track pageViews in every route change. If I don't enable I don't get any pageviews so I suppose I need it.

However on the very first page load (or if I do a hard refresh) I always get two duplicate pageviews instead of one.

My site is running on Universal mode by the way.

Can anyone help me out?

Thanks in advance!

**Comment by proArtex** [comment]
@Damil2020 take a look at my comment here: https://github.com/nuxt-community/gtm-module/issues/33

**Comment by J0XX0** [comment]
@Damil2020 Disable pageview tracking in your nuxt.config.js, and add pageview tag inside Google Tag Manager with the following triggers; "All Pages", "History Change".

Read more here: https://support.google.com/tagmanager/answer/7679322?hl=en

Good luck

**Comment by proArtex** [comment]
@J0XX0 This approach has its own drawbacks. It will mess up meta info from current and previous  route due to vue-meta delays, [see here](https://github.com/nuxt-community/analytics-module/issues/8). I couldn't fix it though. So tracking with 250ms delay works more precise in this case.

**Comment by J0XX0** [comment]
@proArtex I see, that's happening for us too. Since the pageview URL is correct we don't really care for the page title. Facebook Analytics doesn't even seem to collect the page title anyway, so I think it's not really an important unit of measure. 

I would say tracking with 250ms delay just for the title is a big no-no, considering you'd lose data; only in order for the page title to be correct. But whatever works for your case.

**Comment by J0XX0** [comment]
@proArtex you could potentially set custom Javascript in GTM as well to await the title and then push the pageview. Keeps all your analytics on GTM (dynamic) instead of built-in nuxt.

**Comment by nathanchase** [comment]
@J0XX0 @proArtex I'm struggling with the same issue, where the page title is not updated when gtm-module sends the pageTracking pageview event.

I'm not sure exactly where to place a manual pageview event, since it needs to happen on every page - both server-side AND client-side.

I've been trying to figure out if it's a matter of somehow hooking this module directly into vue-meta's [afterNavigation](https://vue-meta.nuxtjs.org/api/#afternavigation) property (that waits for the meta to update), or if it's a matter of listening for the [changed](https://vue-meta.nuxtjs.org/api/#changed) method?

All I know is gtm-module's `pageTracking` is unusable because it never sends the correct `pageTitle` value on a client-side route navigation.

Any thoughts, @pi0 or @manniL ?

**Comment by J0XX0** [comment]
@nathanchase You could push a custom datalayer / pageview event to GTM from the ``mounted()`` function, and set a timer / await for the onroutechange or something. I don't think there is a callback for onTitleChange, but I'm not sure.

For the server-side AND client-side, analytics is client-only. GTM & Analytics SDKs need to generate a clientId and set a cookie using the analytics.js library, so your only way is to implement it on the client side.

**Comment by nathanchase** [comment]
@J0XX0 I should be specific in saying that I need the pageview tracking to occur whether someone lands on a page after a full page browser refresh (server-rendered) or if someone is navigating around the site in-browser (client-side route navigation).

So I just need to ensure the meta info is accurate in both scenarios, and get `gtm-module` to send along the appropriate datalayer object.

**Comment by J0XX0** [comment]
You could create a basic pageview trigger in Google Tag Manager, and from
there on push the custom GTM events from your Nuxt application on route
change. The pageview trigger is called once on the first (server rendered)
visit, and from thereon the onRouteChange event will be pushing pageviews
from nuxt.

**Comment by nathanchase** [comment]
Yeah, I already have GTM listening for the `nuxtRoute` event, and triggering accordingly. But gtm-module has nothing to send: 
![image](https://user-images.githubusercontent.com/5963574/85443592-748ade80-b55f-11ea-8f48-2e8dd5ad356a.png)

I'm pretty sure it's an issue of how this method inside gtm-module is working:
```js
function startPageTracking(ctx) {
  ctx.app.router.afterEach((to) => {
    setTimeout(() => {
      ctx.$gtm.push(to.gtm || {
        routeName: to.name,
        pageType: 'PageView',
        pageUrl: '<%= options.routerBase %>' + to.fullPath,
        pageTitle: (typeof document !== 'undefined' && document.title) || '',
        event: '<%= options.pageViewEventName %>'
      })
    }, 250)
  })
}
```

**Comment by yf-hk** [comment]
Hi guys, is there any working solution now?
